packages/expect/src/utils.js

```diff
@@ -128,8 +128,18 @@ export const iterableEquality = (a: any, b: any) => {
       let allFound = true;
       for (const aValue of a) {
         if (!b.has(aValue)) {
-          allFound = false;
-          break;
+          let has = false;
```

Comment:
I like the look of this :smile: 

I know the current approach is to use the `for...of` loop technique, I find this quite hard to reason about (although not every might agree with me :wink:)

How about something like: 

```js

    ...

    } else if (isA('Set', a) || isImmutableUnorderedSet(a)) {

      const arrayA = [...a];
      return [...b].every(item => a.has(item) || contains(arrayA, item));

    } else if (isA('Map', a) || isImmutableUnorderedKeyed(a)) {

      const arrayA = [...a];
      return [...b].every(
        ([name, item]) =>
          (a.has(name) && equals(item, a.get(name), [iterableEquality])) ||
          contains(arrayA, [name, item]),
      );
    }
```

With a contains function of:
```js
  const contains = (list, value) => {
    return list.findIndex(item => equals(item, value, [iterableEquality])) > -1;
  };
```

I'm not sure if there is any perf difference of the two implementations.

What do you think?

Reply:
I think its fine to leave it as `for...of` actually, it works and we have the tests covering the behaviour if it needs changing in the future :smile: great job with your first contribution dude!

Reply:
Thank you! Also, thanks for the code review. Appreciate it!